File: crates/varde-code/tests/fixtures/python/structural.py
# Structural fixtures: Function, Class, Parameter, MemberAccess, Variable.
# Note: Python has no interface construct (ABCs are runtime classes) — the
# Interface kind is carved out for python (see REQUIRED_KINDS in
# src/extract/langs/python.rs).

def greet(name: str) -> str:
    return f"Hello {name}"


class Greeter:
    greeting: str = ""

    def __init__(self, greeting: str = "hi"):
        self.greeting = greeting

    def greet(self, other: str = "world") -> str:
        return f"Hello {other}"


def compute(a, b=2, *args, c: int, **kwargs):
    total = a + b
    return total

File: crates/varde-code/tests/fixtures/javascript/structural.js
// Structural fixtures: Function, Class, Parameter.
// Note: JavaScript has no interface construct (interface_declaration is a
// TypeScript kind) — the Interface kind is carved out for javascript (see
// REQUIRED_KINDS in src/extract/langs/javascript.rs).

function greet(name) {
  return `Hello ${name}`;
}

function add(a, b = 1, ...rest) {
  const total = a + b;
  return total;
}

class Greeter {
  constructor(greeting) {
    this.greeting = greeting;
  }
}
